### DISCUSSION TITLE
Filter behavior - Is it possible to allow all characters input by a user to be displayed?

### DISCUSSION THREAD

**Discussion by joelkeeling** [Exahilosys/survey#32] [Q&A] [answered]
Hello,
I am trying to build an easy to use application for non technical users. In one of my prompts I am using a select() routine with the text filter. I have noticed when entering characters into the filter, characters are not printed into the filter field unless they have a meaningful impact on results. Unfortunately, this behavior is not intuitive for my non technical users and some of them incorrectly believe the filter is broken... Is there any way to force ALL characters entered to be printed into the filter space regardless of their impact on the results?

I believe these lines from `_search_execute_setup()` in the `Mesh(Mutate)` class is what is causing this behavior:

`if set(old_spots) == set(cur_spots):`
`    raise Error('inconsequential_search_argument', argument = argument)`

I don't see an obvious flag to avoid this check, but thought I'd ask.
Thanks!

**Comment by Exahilosys** [comment, marked as answer]
With [v5.4.0](https://github.com/Exahilosys/survey/releases/tag/v5.4.0) I added the `permit = True` parameter to select and basket that allows search typing to continue without effect when it would have otherwise raised an error internally and prevented the user from typing. With default parameters, the part of the search that is invalid or inconsequential is painted in red:

![image](https://github.com/Exahilosys/survey/assets/22274370/47b19bf7-caa2-41e1-859a-6c0c0a0e3680)

Feel free t o close the discussion if this solves you issue!

**Comment by joelkeeling** [comment]
Thanks for adding this feature! That was insanely fast haha
